Library/Homebrew/standalone/init.rb

```diff
@@ -52,3 +52,15 @@
 $LOAD_PATH.unshift "#{HOMEBREW_LIBRARY_PATH}/vendor/bundle/#{RUBY_ENGINE}/#{Gem.ruby_api_version}/gems/" \
                    "bundler-#{Homebrew::HOMEBREW_BUNDLER_VERSION}/lib"
 $LOAD_PATH.uniq!
+
+# These warnings are nice but often flag problems that are not even our responsibly,
+# including in some cases from other Ruby standard library gems.
+# We strictly only allow one version of Ruby at a time so future compatibility
+# doesn't need to be handled ahead of time.
+if defined?(Gem::BUNDLED_GEMS)
+  [Kernel.singleton_class, Kernel].each do |kernel_class|
+    next unless kernel_class.respond_to?(:no_warning_require, true)
+
+    kernel_class.alias_method :require, :no_warning_require
```

Comment:
Can this use the `Warning` gem instead? Feels like a nicer API if possible.

Reply:
We already did but it never worked. I got spammed with `base64` warnings every time I ran `brew tests`.

Ruby 3.3.5 also makes it significantly unclearer what needs it. Fiddle is one but it seems that `json` and `irb` now need it too despite not being deprecated, because internally they use code that still depends on other deprecated bundled gems.

Reply:
Cool, this approach makes sense then, thanks 👍🏻